src/macho.cpp

```diff
@@ -23,26 +31,53 @@ namespace {
 
 /**************************************************************************************************/
 
-struct section_64 {
-    char sectname[16]{0};
-    char segname[16]{0};
-    std::uint64_t addr{0};
-    std::uint64_t size{0};
-    std::uint32_t offset{0};
-    std::uint32_t align{0};
-    std::uint32_t reloff{0};
-    std::uint32_t nreloc{0};
-    std::uint32_t flags{0};
-    std::uint32_t reserved1{0};
-    std::uint32_t reserved2{0};
-    std::uint32_t reserved3{0};
+struct macho_reader {
```

Comment:
I created a new structure to hold state while the Mach-O files are being read. The goal of this structure is to 1) populate its `dwarf` field, or 2) derive dylib dependencies. The "switch" for which mode the reader is in is based on the callbacks it is given. Since die scanning and dylib scanning are mutually exclusive, the callbacks provided determine which path the `macho_reader` should take.